js/autotrack/common.js

```diff
@@ -1,13 +1,31 @@
+import * as _ from 'lodash';
+
 import { extractProps } from '../util/extractProps';
 import { builtinPropExtractorConfig } from '../propExtractorConfig';
 
+// Returns an object containing a base set of component properties if we're not ignoring the
+// full interaction due to HeapIgnore.
+// Returns null if we're ignoring the full interaction due to HeapIgnore.
 export const getBaseComponentProps = componentThis => {
-  const touchableHierarchy = getComponentHierarchy(componentThis);
+  // Get the hierarchy traversal from root to target component, then get the actual hierarchy from
+  // the traversal representation.
+  const touchableHierarchyTraversal = getComponentHierarchyTraversal(componentThis);
```

Comment:
I'm a bit confused what a "Traversal" is in this sense.  I get that you have to go all the way up and then come back down to cover the ignored bits, but has it materially changed in terms of what you're getting out of the function call?

Reply:
And lol, yes, this question would be MUCH easier to answer in typescript... I'd be in favor of just ripping the band-aid and making this ts happy.

I also think, btw, that the code in this file predates when we started trying ts, so it's possible there's no good reason at all!

Reply:
The [comment describing getFiberNodeComponentHierarchyTraversal return type](https://github.com/heap/react-native-heap/pull/86/files#diff-7e0901a87b208cb50d8e49d8090b3770R49) might help answer your question.

Basically, the function used to return something like
```
AppComponent;[foo=bar];|MyScreen;|MyButton;|
```

Now it returns something like
```
[
  {
    elementName: 'AppComponent',
    fiberNode: <node ref>,
    propsString: '[foo=bar];',
  }, {
    elementName: 'MyScreen',
    ...
  },
  ...
]
```

I'll see about converting this to TS either in this PR or in a followup PR